packages/main/cypress/specs/AINotice.cy.ts

```diff
@@ -0,0 +1,60 @@
+import "../../src/AINotice.js";
+import "../../../icons/src/AllIcons.js";
+
+describe("AINotice component", () => {
+	it("should display attribution text, verification text and icon when all of them are set.", () => {
+		cy.mount("<ui5-ainotice attribution-text='Test Attribution' verification-text='Test Verification' show-icon='true'></ui5-ainotice>");
+		cy.get("ui5-ainotice").shadow().find(".link-text")
+			.should("contain.text", "Test Attribution");
+
+		cy.get("ui5-ainotice").shadow().find(".verificationText")
+			.should("contain.text", "Test Verification");
+
+		cy.get("ui5-ainotice").shadow().find("ui5-link")
+			.should("have.attr", "icon", "sap-icon://ai");
+	});
+
+	it("should display only the icon button if the icon property is set to true and there are no attribution and verification text.", () => {
+		cy.mount("<ui5-ainotice show-icon='true'></ui5-ainotice>");
+
+		cy.get("ui5-ainotice").shadow().find("ui5-button")
+			.should("exist")
+			.should("have.attr", "icon", "sap-icon://ai");
+
+		cy.get("ui5-ainotice").shadow().find(".link-text").should("not.exist");
+		cy.get("ui5-ainotice").shadow().find(".verificationText").should("not.exist");
+	});
+
+	it("should display only attribution text when there is no icon and verification text.", () => {
+		cy.mount("<ui5-ainotice attribution-text='Test Attribution'></ui5-ainotice>");
+
+		cy.get("ui5-ainotice").shadow().find(".link-text")
+			.should("contain.text", "Test Attribution");
+		cy.get("ui5-ainotice").shadow().find(".verificationText")
+			.should("not.exist");
+	});
+
+	it("should display both attribution and verification text when there is no icon.", () => {
+		cy.mount("<ui5-ainotice attribution-text='Test Attribution' verification-text='Test Verification' show-icon='false'></ui5-ainotice>");
```

Comment:
show-icon='false' is actually truthy. if you want to test that show-icon is false, you must not sett the attribute at all